store/src/java/com/zimbra/cs/datasource/imap/ImapFolderSync.java

```diff
@@ -277,6 +279,10 @@ public void syncMessages() throws ServiceException, IOException {
 
         // Restore previously cached sync state or create new state
         syncState = imapSync.removeSyncState(localFolder.getId());
+        int lastChangeId = 0;
+        if (remoteFolder.getPath().equals(DRAFTS) && syncState != null) {
+            lastChangeId = syncState.getLastChangeId();
+        }
```

Comment:
Can we move the 4 lines to around line 359? `lastChangeId` is used there. It is more readable.

Reply:
We require older lastChangeId from older syncstate, and the value of syncState changes at line 287 
hence cannot move this code around 359 as we won't get required value then

Reply:
I got it.